lib/SILGen/SILGenDistributed.cpp

```diff
@@ -244,12 +384,27 @@ void SILGenFunction::emitDistributedActorReady(
   // Only designated initializers get the lifecycle handling injected
   assert(ctor->isDesignatedInit());
 
-  SILValue transport = findFirstDistributedActorSystemArg(F);
+  auto &C = ctor->getASTContext();
+  auto *dc = ctor->getDeclContext();
+  auto classDecl = dc->getSelfClassDecl();
 
   FullExpr scope(Cleanups, CleanupLocation(loc));
   auto borrowedSelf = actorSelf.borrow(*this, loc);
 
-  emitActorReadyCall(B, loc, borrowedSelf.getValue(), transport);
+  // --- load the actor system from the actor instance
+  ManagedValue actorSystem;
+  SGFContext sgfCxt; // TODO: is this right?
+  {
+    VarDecl *property = lookupProperty(classDecl, C.Id_actorSystem);
+    Type formalType = F.mapTypeIntoContext(property->getInterfaceType());
+    SILType loweredType = getLoweredType(formalType).getAddressType();
+    SILValue actorSystemRef = emitActorPropertyReference(
+                                *this, loc, borrowedSelf.getValue(), property);
+    actorSystem = emitLoad(loc, actorSystemRef,
+                           getTypeLowering(loweredType), sgfCxt, IsNotTake);
+  }
```

Comment:
Not sure if this is enough... does this handle a struct system as well? 🤔

Reply:
Hmm... maybe it's right... I was just remembering my trauma from the SIL thunk impl:

```
      auto systemRef = emitActorPropertyReference(
           *this, loc, selfValue.getValue(),
           lookupProperty(selfTyDecl, ctx.Id_actorSystem));

       auto actorSystemTy = systemRef->getType();

       // FIXME: this is wrong for struct with values, and classes?
       // %17 = load %16 : $*FakeActorSystem // users: %21, %20, %18
       SILValue systemLoaded;
       if (actorSystemTy.isAddressOnly(F)) {
         assert(false && "isAddressOnly");
       } else {
         if (actorSystemTy.isAddress()) {
           systemLoaded = B.createTrivialLoadOr(
               loc, systemRef, LoadOwnershipQualifier::Copy);
         } else {
           assert(false);
         }
       }
```

https://github.com/apple/swift/pull/41616/files#diff-375d2a63e22265971efa52970e47979b7414d7f6dc7149100f7a6f9fb7158e0eL826

Reply:
I think because it's using the SILGenFunction's `emitLoad`, it will be smart about whether it's a class or struct and what to do in order to perform the load. That's probably why it seems to be working, but I'll add a test to cover both.

Reply:
Nice, thank you!